packages/apollo-server-core/src/caching.ts

```diff
@@ -1,4 +1,67 @@
+import { ExecutionResult } from 'graphql';
+
 export interface PersistedQueryCache {
   set(key: string, data: string): Promise<any>;
   get(key: string): Promise<string | null>;
 }
+
+export function calculateCacheControlHeaders(
+  responses: Array<ExecutionResult & { extensions?: Record<string, any> }>,
+) {
+  let maxAge = Number.MAX_VALUE;
+  let publicOrPrivate = 'public';
+
+  //Because of the early exit, we are unable to use forEach. While a reduce
```

Comment:
I am hesitant to mention any formatting issues that prettier doesn't deal with automatically, but I've been meaning to mention that your style of not putting a space after `//` seems to be idiosyncratic in our codebases.